### PR TITLE
Aleph and Gun Rebalance

### PR DESCRIPTION
**Pull request by Kitsunemitsu**
## About The Pull Request
Paradise lost now does 60 damage instead of 40
Twilight now does 160 damage instead of 100, It's a fucking boss weapon
Da Capo has it's CD equalized across the board, but only works on a single target.
**SOAS now does 28 instead of 25 damage, and does between 1 and 1.5 damage (based off SP) This Equalizes damage out a bunch.**
Blooming is now 80/70/50 instead of 70/60/40. It had no special abilities aside from changing forms, which was "ok at best" according to player feedback.
Hornet now does 30 red melee, and has 0.3 Fire Delay.
**Magic Bullet now has double the fire delay. If you are wearing the suit, this is halved.**
Loyalty DPS was lowered from 114 to 84
Correctional fire delay reduced to 7
Executive now does 10 pale. We never had a pale gun. It still sucks ass, which is on brand.

All pistols are now bulky, but can fit in an EGO belt.
Added Special text to guns.

## Why It's Good For The Game
A laundry list of buffs and nerfs to many aleph and gun ego.
I've noticed that adoration, twilight and PL weapons were garbage, while SOAS and Magic bullet was too good.
Pistols have also been getting lots of use.

Basically - I buffed weapons I didn't see due to them sucking
Nerfed the guns that I saw TOO much due to them

## Changelog
:cl:
balance: Aleph and Gun rebalance
/:cl:

### CODE REVIEW

File: code/game/objects/items/ego_weapons/aleph.dm [hunk 1]
```diff
@@ -304,7 +305,7 @@
 	desc = "A rose is a rose, by any other name."
 	special = "Use this weapon to change it's damage type between red, white and pale."	//like a different rabbit knife. No black though
 	icon_state = "rosered"
-	force = 70 //Less damage, can swap damage type
+	force = 80 //Less damage, can swap damage type
```
Comment (EgorDinamit) [inline code/game/objects/items/ego_weapons/aleph.dm:308]: 80 pale damage is bullshit and too much. TOO much, Kirie.

File: code/modules/projectiles/guns/ego_gun/aleph.dm [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
 	inhand_icon_state = "star"
+	special = "This gun fires 240 rounds per minute. \
+			This gun scales with remaining SP."
 	ammo_type = /obj/item/ammo_casing/caseless/ego_star
 	weapon_weight = WEAPON_HEAVY
 	spread = 5
 	fire_sound = 'sound/weapons/ego/star.ogg'
 	vary_fire_sound = TRUE
-	fire_sound_volume = 50
+	fire_sound_volume = 25
```
Comment (EgorDinamit) [inline code/modules/projectiles/guns/ego_gun/aleph.dm:13]: But why :thinking:

File: code/modules/projectiles/guns/ego_gun/he.dm [hunk 3]
```diff
@@ -3,10 +3,17 @@
...
 	force = 20
 	damtype = BLACK_DAMAGE
 	armortype = BLACK_DAMAGE
 	fire_delay = 3
 	fire_sound = 'sound/weapons/gun/rifle/shot.ogg'
+
+
+/obj/item/gun/ego_gun/prank/EgoAttackInfo(mob/user)
+	if(chambered && chambered.BB)
+		return "<span class='notice'>Its bullets deal [chambered.BB.damage] [chambered.BB.damage_type] randomly chosen damage, and [force] [damtype] melee damage</span>"
```
Comment (EgorDinamit) [inline code/modules/projectiles/guns/ego_gun/he.dm:18]:
```suggestion
		return "<span class='notice'>Its bullets deal [chambered.BB.damage] [chambered.BB.damage_type]damage, and [force] [damtype] melee damage</span>"
```

File: code/modules/projectiles/guns/ego_gun/waw.dm [hunk 4]
```diff
@@ -56,25 +68,39 @@
 	The weapon's bullets travel across the corridor, along the horizon."
 	icon_state = "magic_bullet"
 	inhand_icon_state = "magic_bullet"
+	special = "This weapon fires extremely slowly. \
+		This weapon pierces all targets. \
+		This weapon fires significantly faster wearing the matching armor"
 	ammo_type = /obj/item/ammo_casing/caseless/ego_magicbullet
 	weapon_weight = WEAPON_HEAVY
-	fire_delay = 12
+	fire_delay = 32	//Put on the armor, jackass.
```
Comment (EgorDinamit) [inline code/modules/projectiles/guns/ego_gun/waw.dm:76]: This might be too much, set to 24.

File: code/modules/projectiles/projectile/ego_bullets/aleph.dm [hunk 5]
```diff
@@ -19,6 +19,6 @@
 	if(!isbot(H) && isliving(H))
 		H.visible_message("<span class='warning'>[target] is hit by [src], they seem to wither away!</span>")
 		for(var/i = 1 to 10)
-			addtimer(CALLBACK(H, /mob/living/proc/apply_damage, rand(4,6), BLACK_DAMAGE, null, H.run_armor_check(null, BLACK_DAMAGE)), 2 SECONDS * i)
+			addtimer(CALLBACK(H, /mob/living/proc/apply_damage, rand(6,8), BLACK_DAMAGE, null, H.run_armor_check(null, BLACK_DAMAGE)), 2 SECONDS * i)
```
Comment (EgorDinamit) [inline code/modules/projectiles/projectile/ego_bullets/aleph.dm:22]: That's +20 damage.

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Da capo changes are rather needless and will only clutter the code.

File: code/game/objects/items/ego_weapons/aleph.dm (hunk 1 above)
Comment (Kitsunemitsu) [reply code/game/objects/items/ego_weapons/aleph.dm:308]: Its not 80 pale, its 80 red, 70 white and 50 pale.
Still kinda meh overall because it doesnt do fuckloads of damage like Justitia and Da Capo

File: code/modules/projectiles/guns/ego_gun/aleph.dm (hunk 2 above)
Comment (Kitsunemitsu) [reply code/modules/projectiles/guns/ego_gun/aleph.dm:13]: It hurt my ears with how loud it was, while everything else was normal

File: code/modules/projectiles/projectile/ego_bullets/aleph.dm (hunk 5 above)
Comment (Kitsunemitsu) [reply code/modules/projectiles/projectile/ego_bullets/aleph.dm:22]: It's actually a slow firing weapon.
SOAS can do 116 damage per second at MINIMUM. 
Adoration with it's slow firing still loses out in dps to it.

**Comment by Kitsunemitsu** [comment]
I like Da Capo being a musical based weapon, where you "Waltz" with a designated partner for increased damage

**Comment by EgorDinamit** [comment]
This just nerfs the weapon, making it useless after you hit someone once.

**Comment by Kitsunemitsu** [comment]
It actually isn't a nerf considering I sped it ip in exchange....

**Comment by Kitsunemitsu** [comment]
Egor, Da Capo is NOT nerfed. 
It does 80 DPS, the same as mimicry, but it gets a 15% boost if you're attacking the same target.

**Comment by Kitsunemitsu** [comment]
After some testing and community feedback, I have gotten generally great feedback on Da Capo, and it no longer being a Justitia Clone.

File: code/game/objects/items/ego_weapons/aleph.dm [hunk 6]
```diff
@@ -130,20 +126,25 @@
...
 	var/combo_wait = 14
+	var/waltz_partner
+	//I'm making Da Capo a waltzing weapon, It should play like a rhythm game. - Kirie.
 
 /obj/item/ego_weapon/da_capo/attack(mob/living/M, mob/living/user)
 	if(!CanUseEgo(user))
 		return
 	if(world.time > combo_time)
 		combo = 0
+	if(!waltz_partner||waltz_partner!=M)
```
Comment (EgorDinamit) [inline code/game/objects/items/ego_weapons/aleph.dm:137]:
```suggestion
	if(!waltz_partner || waltz_partner != M)
```
Le quality control.

File: code/game/objects/items/storage/belt.dm [hunk 7]
```diff
@@ -745,3 +745,26 @@
 		/obj/item/shovel/spade,
 		/obj/item/gun/energy/floragun
 		))
+
+/obj/item/storage/belt/ego
+	name = "ego small arms belt" //Carn: utility belt is nicer, but it bamboozles the text parsing.
+	desc = "Holds various small ego weapons."
```
Comment (EgorDinamit) [inline code/game/objects/items/storage/belt.dm:751]: It is specifically for pistols, so might be worth mentioning it here.

File: code/modules/jobs/job_types/agent.dm [hunk 8]
```diff
@@ -95,7 +95,8 @@
 	name = "Agent"
 	jobtype = /datum/job/agent
 
-	belt = /obj/item/pda/security
+	belt = /obj/item/storage/belt/ego
```
Comment (EgorDinamit) [inline code/modules/jobs/job_types/agent.dm:98]: What's the point of making it spawn on people? I'd suggest putting it on the maps, or adding a vendor for all this additional equipment.

File: code/modules/projectiles/guns/ego_gun/waw.dm [hunk 9]
```diff
@@ -20,21 +20,32 @@
...
+	damtype = RED_DAMAGE
+	armortype = RED_DAMAGE
 
 	attribute_requirements = list(
 							FORTITUDE_ATTRIBUTE = 60
 							)
 
+/obj/item/gun/ego_gun/hornet/EgoAttackInfo(mob/user)
+	if(chambered && chambered.BB)
+		return "<span class='notice'>Its bullets deal [chambered.BB.damage] [chambered.BB.damage_type] damage, and [force] [damtype] melee damage</span>"
```
Comment (EgorDinamit) [inline code/modules/projectiles/guns/ego_gun/waw.dm:38]:
```suggestion
		return "<span class='notice'>Its bullets deal [chambered.BB.damage] [chambered.BB.damage_type] damage. Weapon does [force] [damtype] damage in melee.</span>"
```

File: code/modules/projectiles/guns/ego_gun/waw.dm [hunk 10]
```diff
@@ -56,25 +67,39 @@
...
+		return FALSE
+	if (!istype(user,/mob/living/carbon/human))
+		return
+	fire_delay = 24
+	var/mob/living/carbon/human/myman = user
+	var/obj/item/clothing/suit/armor/ego_gear/magicbullet/Y = myman.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+	if(istype(Y))
+		fire_delay = 16
+	..()
+
```
Comment (EgorDinamit) [inline code/modules/projectiles/guns/ego_gun/waw.dm:93]:
```suggestion
/obj/item/gun/ego_gun/magicbullet/before_firing(atom/target, mob/user)
	fire_delay = 24
	var/mob/living/carbon/human/myman = user
	var/obj/item/clothing/suit/armor/ego_gear/magicbullet/Y = myman.get_item_by_slot(ITEM_SLOT_OCLOTHING)
	if(istype(Y))
		fire_delay = 16

```

**Review by EgorDinamit — CHANGES_REQUESTED** [review]

File: code/modules/jobs/job_types/agent.dm (hunk 8 above)
Comment (Kitsunemitsu) [reply code/modules/jobs/job_types/agent.dm:98]: It was mostly for testing because this was TMed for a while. I will add to the maps.

File: code/modules/jobs/job_types/agent.dm [hunk 11]
```diff
@@ -94,7 +94,6 @@
 /datum/outfit/job/agent
 	name = "Agent"
 	jobtype = /datum/job/agent
-
```
Comment (EgorDinamit) [inline code/modules/jobs/job_types/agent.dm:97]: As funny as it is, can you revert this change?

Comment (Kitsunemitsu) [reply code/modules/jobs/job_types/agent.dm:97]: Can do,
